src/sentry/static/sentry/app/components/organizations/multipleEnvironmentSelector.jsx

```diff
@@ -57,6 +58,25 @@ class MultipleEnvironmentSelector extends React.PureComponent {
     };
   }
 
+  componentDidUpdate(prevProps, prevState) {
+    // When projects change we may need to update the selected value if the current values
+    // become invalid.
+    const sharedProjects = intersection(
+      prevProps.selectedProjects,
+      this.props.selectedProjects
+    );
+    if (sharedProjects.length !== prevProps.selectedProjects.length) {
```

Comment:
Do we only compare the length to determine if a change has occurred? What if the actual projects have changed? Should the check be like `newprojects.issubset(oldprojects)` instead?

Reply:
I was checking the length because comparing the arrays themselves requires sorting both arrays as `[14, 11]` and `[11, 14]` are not equal. If the two lists are the same, then the intersection of new + old should contain the old list. This also skips a state update where the new selection contains the old selection and only adds additional projects.

Reply:
Ah, oops